internal/service/advancedclustertpf/resource_compatibility_reuse.go

```diff
@@ -0,0 +1,90 @@
+package advancedclustertpf
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cast"
+	admin20240530 "go.mongodb.org/atlas-sdk/v20240530005/admin"
+	"go.mongodb.org/atlas-sdk/v20241113001/admin"
+)
+
+func FormatMongoDBMajorVersion(version string) string {
+	if strings.Contains(version, ".") {
+		return version
+	}
+	return fmt.Sprintf("%.1f", cast.ToFloat32(version))
+}
+
+func getReplicationSpecIDsFromOldAPI(ctx context.Context, projectID, clusterName string, api admin20240530.ClustersApi) (map[string]string, error) {
+	clusterOldAPI, _, err := api.GetCluster(ctx, projectID, clusterName).Execute()
+	if apiError, ok := admin20240530.AsError(err); ok {
+		if apiError.GetErrorCode() == "ASYMMETRIC_SHARD_UNSUPPORTED" {
+			return nil, nil // if its the case of an asymmetric shard an error is expected in old API, replication_specs.*.id attribute will not be populated
+		}
+		readErrorMsg := "error reading  advanced cluster with 2023-02-01 API (%s): %s"
+		return nil, fmt.Errorf(readErrorMsg, clusterName, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("error reading  advanced cluster with 2023-02-01 API (%s): %s", clusterName, err)
```

nit: I think we can directly use the string as we normally do

Reply:
SGTM